build/tutorials/platform/port-your-ethereum-dapp.md

```diff
@@ -0,0 +1,145 @@
+
+### Hardhat
+
+Hardhat is the newest development and testing environment for Solidity smart contracts, and the one our developers use the most. Due to its superb testing support, it is the recommended way of developing for Avalanche.
+
+[Here](https://github.com/ava-labs/avalanche-smart-contract-quickstart) is a quickstart repository that our developers use to start new projects.
+
+## Avalanche Explorer
+
+Essential part of the smart contract development environment is the explorer, that indexes and serves blockchain data. Main net C-Chain explorer is available at [https://cchain.explorer.avax.network/](https://cchain.explorer.avax.network/) and test net explorer at [https://cchain.explorer.avax-test.network](https://cchain.explorer.avax-test.network/). Besides the web interface, it also exposes the standard [Ethereum JSON RPC API](https://eth.wiki/json-rpc/API).
+
+## Avalanche Faucet
+
+For development purposes, you will need test tokens. Avalanche has a [Faucet](https://faucet.avax-test.network/) that drips tet tokens to address of your choice. Paste your C-Chain address there.
+
+If you need, you can also run a faucet locally, but building it from the [repository](https://github.com/ava-labs/avalanche-faucet).
+
+## Contract verification
+
+Smart contract verification provides transparency for users interacting with smart contracts by publishing the source code, allowing everyone to attest that it really does what it claims to do. You can verify your smart contracts using the [C-Chain explorer](https://cchain.explorer.avax.network/). The procedure is simple:
+* navigate to your published contract address on the explorer
+* on the `code` tab select `verify & publish`
+* copy and paste the flattened source code and enter all the build parameters exactly as they are on the published contract
+* click `verify & publish`
+
+If successful, the `code` tab will now have a green checkmark, and your users will be able to verify the contents of your contract. This is a strong positive signal that your users can trust your contracts, and it is strongly recommended for all production contracts.
+
+## Gotchas and things to look out for
+
+Avalanche Platform's C-Chain is EVM-compatible, but it is not identical. There are some differences you need to be aware of, otherwise you may create subtle bugs or inconsistencies in how you dapps behave.
+
+Here are the main differences you should be aware of.
+
+### Measuring time
+
+It is customary on Ethereum to use block height progress as a proxy for time. You should not do that on Avalanche. Chains on Avalanche are quiescent, meaning that if there is no activity, there are no blocks produced. The opposite is also true, if there is great amount of activity, blocks are produced very fast. Because of that, you should not measure the passage of time by the amount of blocks that are produced. You both get inaccurate results, and be open to manipulation by third parties.
```

Comment:
The last sentence in this paragraph isn't grammatically correct. I'd recommend: The results will not be accurate, and your contract may be manipulated by third parties.